### PR TITLE
feat: support excalidraw file

### PR DESCRIPTION
**Pull request by hweihwang**
## Summary
- Support new excalidraw file type

### CODE REVIEW

File: lib/private/Repair/RepairMimeTypes.php [hunk 1]
```diff
@@ -334,5 +342,9 @@ public function run(IOutput $out) {
 		if (version_compare($ocVersionFromBeforeUpdate, '29.0.0.10', '<') && $this->introduceReStructuredTextFormatType()) {
 			$out->info('Fixed ReStructured Text mime type');
 		}
+
+		if (version_compare($ocVersionFromBeforeUpdate, '29.0.0.10', '<') && $this->introduceExcalidrawType()) {
```
Comment (juliushaertl) [inline lib/private/Repair/RepairMimeTypes.php:346]:
```suggestion
		if (version_compare($ocVersionFromBeforeUpdate, '30.0.0.0', '<') && $this->introduceExcalidrawType()) {
```
As we are now on 30

File: lib/private/Repair/RepairMimeTypes.php [hunk 2]
```diff
@@ -106,6 +106,14 @@ private function updateMimetypes($updatedMimetypes) {
 		return $count;
 	}
 
+	private function introduceExcalidrawType() {
```
Comment (blizzz) [inline lib/private/Repair/RepairMimeTypes.php:109]: would be an opportunity to add return type hint here and in `updateMimetypes()`

**Review by juliusknorr — APPROVED** [review]

**Comment by juliushaertl** [comment]
Tests needs some adjustment it seems:

> 1) Test\IntegrityCheck\CheckerTest::testVerifyCoreSignatureWithModifiedMimetypelistSignatureData

**Comment by juliushaertl** [comment]
@hweihwang Pushed a small commit to adjust the icon color to match the others.

**Comment by juliushaertl** [comment]
Looks good, cypress seems unrelated, could you squash your commits and rebase to latest master afterwards? Will push my icon fix again then afterwards.

**Comment by juliushaertl** [comment]
@hweihwang Had another conflict in the mean time, could you check?

**Review by max-nextcloud — APPROVED** [review]

**Pull request merged**